File: Server API - luis/probe-server/src/main/java/pt/solvit/probe/server/model/enums/UserProfile.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package pt.solvit.probe.server.model.enums;

/**
 *
 * @author AnaRita
 */
public enum UserProfile {
    ADMIN("Admin", 2),
    SUPER_USER("Super User", 1),
    NORMAL_USER("Normal User", 0);

    private final String name;
    private final int level;

    private UserProfile(String name, int level) {
        this.name = name;
        this.level = level;
    }

    public String getName() {
        return name;
    }

    public int getLevel() {
        return level;
    }
}

File: Client/my-app/src/app/Model/UserProfile.ts
export enum UserProfile{
    NORMAL_USER = 0,
    SUPER_USER = 1,
    ADMIN = 2
}

export namespace UserProfile{
    export function getValueFromString(profile : string) : number{
        //return 
        //console.log("In UserProfile, profile is: " + UserProfile[profile])
        switch(profile){
            case "NORMAL_USER": return UserProfile.NORMAL_USER
            case "SUPER_USER": return UserProfile.SUPER_USER
            case "ADMIN": return UserProfile.ADMIN
            default: throw "UserProfile doesn't exist"
        }
    }
}
